src/test/tests/applicabilityScan.test.ts

```diff
@@ -0,0 +1,328 @@
+import { PackageType } from '../../main/types/projectType';
+import { DependencyScanResults } from '../../main/types/workspaceIssuesDetails';
+import { AnalyzerUtils } from '../../main/treeDataProviders/utils/analyzerUtils';
+import { DependencyIssuesTreeNode } from '../../main/treeDataProviders/issuesTree/descriptorTree/dependencyIssuesTreeNode';
+import { Severity } from '../../main/types/severity';
+import { CveTreeNode } from '../../main/treeDataProviders/issuesTree/descriptorTree/cveTreeNode';
+import { FileTreeNode } from '../../main/treeDataProviders/issuesTree/fileTreeNode';
+import { CodeFileTreeNode } from '../../main/treeDataProviders/issuesTree/codeFileTree/codeFileTreeNode';
+import { IEvidence } from 'jfrog-ide-webview';
+import { CodeIssueTreeNode } from '../../main/treeDataProviders/issuesTree/codeFileTree/codeIssueTreeNode';
+import { ApplicableTreeNode } from '../../main/treeDataProviders/issuesTree/codeFileTree/applicableTreeNode';
+import { removeWindowsWhiteSpace } from './utils/utils.test';
+
+let logManager: LogManager = new LogManager().activate();
+
+describe('Contextual Analysis Scan Tests', () => {
+    const scanApplicable: string = path.join(__dirname, '..', 'resources', 'applicableScan');
+    let tempFolder: string = ScanUtils.createTmpDir();
+
+    [
+        {
+            name: 'One root',
+            file: path.join(scanApplicable, 'requestOneRoot.yaml'),
+            roots: ['/path/to/root'],
+            cves: ['CVE-2021-3918'],
+            skip: []
+        },
+        {
+            name: 'Multiple roots',
+            file: path.join(scanApplicable, 'requestMultipleRoots.yaml'),
+            roots: ['/path/to/root', '/path/to/other'],
+            cves: ['CVE-2021-3918', 'CVE-2021-3807', 'CVE-2022-25878'],
+            skip: ['/path/to/skip']
+        }
+    ].forEach(test => {
+        it('Check generated Yaml request - ' + test.name, () => {
+            let request: ApplicabilityScanArgs = getApplicabilityScanRequest(test.roots, test.cves, test.skip);
+            let actualPath: string = path.join(tempFolder, test.name);
+            fs.writeFileSync(actualPath, getDummyRunner().requestsToYaml(request));
+            assert.deepEqual(
+                fs.readFileSync(actualPath, 'utf-8').toString(),
+                removeWindowsWhiteSpace(fs.readFileSync(test.file, 'utf-8').toString())
+            );
+        });
+    });
+
+    describe('Analyzer run fail tests', () => {
+        let response: ApplicabilityScanResponse;
+
+        before(() => {
+            response = getDummyRunner().generateResponse(undefined);
+        });
+
+        it('Check response defined', () => {
+            assert.isDefined(response);
+        });
+
+        it('Check response attributes', () => {
+            assert.isUndefined(response.scannedCve);
+            assert.isUndefined(response.applicableCve);
+        });
+    });
+
+    describe('Populate applicable information tests', () => {
```

Comment:
We should also add tests for unknown applicable Cve

Reply:
unknown? 
we have to provide the list of the CVE to scan so we have to have a specific CVE to run on.
In addition there are only two results to the scanner:
fail = applicable issue (checked under describe 'Applicable Cve tests')
pass = not applicable issue (checked under describe 'Not applicable Cve tests')